src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -36,84 +36,26 @@
         int i = 0;
         List<Pet> pets = new LinkedList<Pet>();
         Pet pet;
+        CSVContainer csvContainer = new CSVContainer(csv);
 
-        do {
+        for (int lineIndex = 0; lineIndex < csvContainer.getNumberOfLines(); ++lineIndex) {
```

Comment:
when you refactor a long method, try to extract (small) methods for each intention right at the beginning of your refactoring session.
you can also have a look here, which is basically a copy of the contents of Martin Fowler's Refactoring book: [https://refactoring.guru/extract-method](https://refactoring.guru/extract-method)